src/main/java/clevernotbot/TaskList.java

```diff
@@ -0,0 +1,110 @@
+package clevernotbot;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Represents a list for Task. A <code>TaskList</code> object corresponds to
+ * an ArrayList of task.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int size;
```

Comment:
likewise, possible to make it private final? Maybe create a new TaskList object with each function call